### PR TITLE
parser: fix create table without name

### LINKED ISSUE #29: parser: create table allows incorrect production
**Describe the bug**
`CREATE TABLE(n,FOREIGN KEY()REFERENCES n ON DELETE CASCADE)` can be parsed, and has no table name.
I don't know if the statement is otherwise incorrect, I do assume so.
However, this passes the parser and causes a nil dereference in the compiler.

**Expected behavior**
A parse error.

### PR DESCRIPTION
**Pull request by Shivam010**
Closes #29

### CODE REVIEW

**Comment by Shivam010** [comment]
@xqueries/developer 
Also, I think we should discuss #51 first

**Review by tsatke — CHANGES_REQUESTED** [review]
Please add test cases that prove that the changes work as intended

File: internal/parser/issue29_test.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+func TestIssue29(t *testing.T)  {
+	NegativeTest{
+		Name:  "issue29",
+		Query: "CREATE TABLE(n,FOREIGN KEY()REFERENCES n ON DELETE CASCADE)",
+	}.Run(t)
+}
+
+func TestIssue29WithoutTableName(t *testing.T) {
+	NegativeTest{
+		Name:  "issue29",
```
Comment (tsatke) [inline internal/parser/issue29_test.go:14]: You are already giving a better name in the method name, please use it here as well
```suggestion
		Name:  "issue29 without table name",
```

File: internal/parser/issue29_test.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		Query: "CREATE TABLE (foo)",
+	}.Run(t)
+}
+
+func TestIssue29WithNumericColumn(t *testing.T)  {
+	NegativeTest{
+		Name:  "issue29",
```
Comment (tsatke) [inline internal/parser/issue29_test.go:21]: same here

```suggestion
		Name:  "issue29 with numeric column",
```

File: internal/parser/issue29_test.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+		Query: "CREATE TABLE foo(1)",
+	}.Run(t)
+}
+
+func TestIssue29WithNumericTable(t *testing.T)  {
+	NegativeTest{
+		Name:  "issue29",
```
Comment (tsatke) [inline internal/parser/issue29_test.go:28]:
```suggestion
		Name:  "issue29 with numeric column",
```

**Review by tsatke — CHANGES_REQUESTED** [review]

**Review by tsatke — APPROVED** [review]

**Pull request merged**